web/src/components/storage/ProposalSettingsSection.jsx

```diff
@@ -448,6 +448,65 @@ const EncryptionSettingsForm = ({
   );
 };
 
+/**
+ * Allows to define snapshots enablement
+ * @component
+ *
+ * @param {object} props
+ * @param {ProposalSettings} props.settings - Settings used for calculating a proposal.
+ * @param {onChangeFn} [props.onChange=noop] - On change callback
+ *
+ * @callback onChangeFn
+ * @param {object} settings
+ */
+const SnapshotsField = ({
+  settings,
+  onChange = noop
+}) => {
+  const rootVolume = settings.volumes.find((i) => i.mountPath === "/");
+
+  const initialChecked = rootVolume !== undefined && rootVolume.fsType === "Btrfs" && rootVolume.snapshots;
+  const [isChecked, setIsChecked] = useState(initialChecked);
+
+  if (rootVolume === undefined) { // no root volume is probably some error or still loading
+    return <Skeleton width="25%" />;
+  }
+
+  const switchState = (checked) => {
+    setIsChecked(checked);
+    onChange({ value: checked, settings });
+  };
+
+  const explanation = _("Uses btrfs for the root file system allowing to boot to a previous version of the system after configuration changes or software upgrades.");
+  const selector = (
+    <div>
+      <Switch
+        id="snapshots"
+        label={_("Use Btrfs Snapshots")}
+        isReversed
+        isChecked={isChecked}
+        onChange={switchState}
+      />
+      <div>
+        {explanation}
+      </div>
+    </div>
+  );
+  const enabledStatement = (
+    <div>
+      {_("Btrfs snapshots required by product.")}
+    </div>
+  );
+
+  if (rootVolume.outline.snapshotsConfigurable) {
+    return selector;
+  } else if (rootVolume.fsType === "Btrfs" && rootVolume.snapshots) {
+    return enabledStatement;
+  } else { // strange situation, should not happen
+    return (<></>);
+  }
```

Comment:
Since [React 18](https://github.com/facebook/react/blob/main/CHANGELOG.md#react-2), this is no longer needed.

Now React allows returning `undefined`, and this is the default return value from a JavaScript function when you returns nothing. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function#description.